src/qml/Tracking.qml

```diff
@@ -175,140 +175,236 @@ Item{
-                anchors {
-                    margins: 4
-                    topMargin: 35// Leave space for the toolbar
-                }
+                columns: 2
+                columnSpacing: 0
+                rowSpacing: 5
+
 
-                CheckBox {
-                    id: timeIntervalCheck
-                    text: qsTr( 'Time interval (s)' )
+                Label {
+                    text: qsTr("Activate time interval constraint")
                     font: Theme.defaultFont
+                    wrapMode: Text.WordWrap
+                    Layout.fillWidth: true
+
+                    MouseArea {
+                        anchors.fill: parent
+                        onClicked: timeInterval.toggle()
+                    }
+                }
 
+                QfSwitch {
+                    id: timeInterval
+                    Layout.preferredWidth: implicitContentWidth
+                    Layout.alignment: Qt.AlignTop
+                    checked: positioningSettings.trackerTimeIntervalConstraint
+                    onCheckedChanged: {
+                        positioningSettings.trackerTimeIntervalConstraint = checked
+                    }
+                }
+
+                Label {
+                    text: qsTr("Minimum time [sec]")
+                    font: Theme.defaultFont
+                    wrapMode: Text.WordWrap
+                    enabled: timeInterval.checked
+                    visible: timeInterval.checked
+                    Layout.leftMargin: 8
                     Layout.fillWidth: true
-                    indicator.height: 16
-                    indicator.width: 16
-                    indicator.implicitHeight: 24
-                    indicator.implicitWidth: 24
                 }
 
                 TextField {
-                    id: timeIntervalText
-                    enabled: timeIntervalCheck.checked
-                    height: fontMetrics.height + 20
-                    topPadding: 10
-                    bottomPadding: 10
-                    Layout.fillWidth: true
+                    id: timeIntervalValue
+                    width: timeInterval.width
                     font: Theme.defaultFont
-                    text: '30'
+                    enabled: timeInterval.checked
+                    visible: timeInterval.checked
+                    horizontalAlignment: TextInput.AlignHCenter
+                    Layout.preferredWidth: 60
+                    Layout.preferredHeight: font.height + 20
 
                     inputMethodHints: Qt.ImhFormattedNumbersOnly
+                    validator: DoubleValidator { locale: 'C' }
 
-                    validator: IntValidator {
+                    background: Rectangle {
+                      y: parent.height - height - parent.bottomPadding / 2
+                      implicitWidth: 120
+                      height: parent.activeFocus ? 2: 1
+                      color: parent.activeFocus ? '#4CAF50' : '#C8E6C9'
```

Comment:
These colors are used in the other PR too, definitely need to go now or in a follow up PR into theme.

Reply:
@suricactus , I was about to answer on the other PR. It's used in a fair amount of places. I'm deferring handling these on some exciting feature in a to-do list that will hopefully happen sooner than later :)